Scripts/Python/step_response_identification.py

```diff
@@ -0,0 +1,122 @@
+import pandas as pd
+import numpy as np
+from typing import Tuple
+import control
+from control.matlab import step
+import matplotlib.pyplot as plt
+import easygui
+
+def check_data(dataframe):
+    """
+    Check if the DataFrame represents a step response.
+
+    Args:
+        dataframe (pd.DataFrame): The DataFrame to check.
+
+    """
+
+    assert 'H1_ADD_NOISE' in dataframe.columns, "Invalid experiment CSV file"
+
+    dataframe_test = dataframe.copy()
+    dataframe_test.drop_duplicates(subset='H1_ADD_NOISE', keep='first', inplace=True)
+    nonzero_points = dataframe_test[dataframe_test['H1_ADD_NOISE'] != 0]
+    assert len(nonzero_points) == 1, "The experiment is not a step response"
+
+
+def load_data(file_path: str) -> pd.DataFrame:
+    """
+    Load data from a CSV file.
+
+    Args:
+        file_path (str): The path to the CSV file.
+
+    Returns:
+        pd.DataFrame: The loaded data as a pandas DataFrame.
+    """
+    dataframe = pd.read_csv(file_path)
+    check_data(dataframe)
+    dataframe.set_index('TIME', inplace=True)
+    starting_time = dataframe.loc[dataframe['H1_ADD_NOISE'] > 0.0].index.values[0]
+    dataframe = dataframe.loc[starting_time:]
+    dataframe.index = (dataframe.index - starting_time) / 1000
+    return dataframe
+
+def calculate_parameters(dataframe: pd.DataFrame) -> Tuple[float, float, float, float, float]:
+    """
+    Calculate system parameters and extract relevant data.
+
+    Args:
+        dataframe (pd.DataFrame): The input data as a pandas DataFrame.
+
+    Returns:
+        Tuple[float, float, float, float, pd.DataFrame]: A tuple containing system parameters
+        (K, L, TAU, P, offset)
+    """
+
+    offset = dataframe['H1_TEMP'].values[0]
+    dataframe.loc[:,'H1_TEMP'] = dataframe['H1_TEMP'] - offset
+    window = int(dataframe.index.max() / 10)
+    rolling_mean = dataframe.rolling(window).mean()
+    max_value = rolling_mean.max()[0]
+
+    P = dataframe.loc[dataframe['H1_ADD_NOISE'] > 0]['H1_ADD_NOISE'].values[0]
+    K = dataframe.loc[dataframe['H1_TEMP'] > max_value * 0.99]['H1_TEMP'].mean() / P
+    L = dataframe.loc[dataframe['H1_TEMP'] > 0.0].index.values[0]
+    TAU = dataframe.loc[dataframe['H1_TEMP'] > (0.632 * max_value)].index.values[0]
```

Comment:
This is WRONG. Your TAU is greater than it should be, since you are (implicitly) including L inside of it. Note that the time-constant TAU is the point where the response reaches 63.2% of the final value starting from the time-delay L, and not the beginning of the experiment. Therefore, the correct value here is given by:
TAU = dataframe.loc[dataframe['H1_TEMP'] > (0.632 * max_value)].index.values[0] - L